locale/en/docs/topics/async.md

```diff
@@ -0,0 +1,461 @@
+time and waiting for each to complete before starting the next.
+
+By using asynchronous programming, well-written node.js processes can handle
+hundreds or thousands of requests per second. Fewer processes doing more work
+can be more efficient than many processes waiting idly for external sources to
+return.
+
+
+
+Async Approach: Callbacks
+-------------------------
+In this approach, when a function is called it is passed a function object as
+an argument. Once the result is available it is passed to this function
+object. The function object is often called a "callback" since it is "called"
+to return the result "back" to the caller.
+
+Consider the following example:
+
+```js
+function add(a, b, callback) {
+  setTimeout(function() {
+    var c = a + b;
+    callback(c);
+  }, 100);
+}
+
+add(12, 5, function(sum) {
+  console.log(sum);
+});
+```
+
+When we call `add()` we pass an anonymous function as the third argument. Once
+the sum is available the anonymous function will be called with it.
+
+There are two common refinements to the above technique: events and streams.
+
+
+### Events
+Although a callback is often called once to return the result of an operation,
+there is another common usecase: responding to an event (which might happen
+more than once). In this approach a callback is registered so that it is
+called when something interesting happens later.
+
+```js
+process.on('uncaughtException', function(err) {
+  console.error('TROUBLE!', err.stack);
+});
+```
+
+The callback in this case is sometimes called a "listener" or an "event
+handler". The documentation for each event should describe the signature of
+the callback.
+
+For more information see the [events API][] documentation.
+
+
+### Streams
+Streams are one way of connecting a data producer with a data consumer so that
+data is processed asynchronously. Much of the API is built on events.
+
+For more information see the [stream API][] documentation.
+
+
+### Nodeback
+In node.js there is a common pattern for using callbacks where:
+
+* The callback is the last argument of the function.
+* The first argument of the callback is an error object. (see "Throwing
+  and try/catch" below.)
+
+Example:
+
+```js
+require('fs').readFile('foo.txt', function(err, contents) {
+  if (err) {
+    // somehow respond to the error
+  } else {
+    // use the contents of the file
+  }
+});
+```
+
+This approach to callbacks is so common in node.js that is has been nicknamed
+"nodeback" 😄
+
+
+
+Async Approach: Promises
+------------------------
+In this approach a function returns an object which represents a "promise" to
+provide the actual result at some later time. The [Promise API][] specifies
+how that result becomes available.
+
+The Promise API is an official part of the javascript language starting with
+ECMAScript 2015 (ES6), which is available starting with node.js version 0.12.
+If you're using an earlier version of node.js you can use one of the many
+[promise modules][].
+
```

Comment:
It might be worth it to have a brief primer on Promises here. Or a simple example. Or both. It's just a tad vague compared to callbacks.

If not, maybe link to one?